incubator/hnc/test/conversion/conversion_test.go

```diff
@@ -232,7 +232,57 @@ spec:
 
 		// Verify default types in the new version.
 		FieldShouldContainWithTimeout(configCRD, "", configSingleton, ".apiVersion", "v1alpha2", crdConversionTime)
-		FieldShouldContainMultiple(configCRD, "", configSingleton, ".spec.types", []string{"Role", "RoleBinding"})
+		FieldShouldContainMultiple(configCRD, "", configSingleton, ".spec.types", []string{"roles", "rolebindings"})
+	})
+
+	It("should convert HNCConfig types", func() {
+		// Create a tree with A as the root and B as the child
+		createSampleV1alpha1Tree()
+		// Delete the webhook to apply unsupported modes in v1alpha1.
+		MustRun("kubectl delete validatingwebhookconfigurations.admissionregistration.k8s.io hnc-validating-webhook-configuration")
+		// Set 'propagate', 'remove', unknown ('ignore') modes in v1alpha1
+		cfg := `# temp file created by conversion_test.go
+apiVersion: hnc.x-k8s.io/v1alpha1
+kind: HNCConfiguration
+metadata:
+  name: config
+spec:
+  types:
+  - apiVersion: rbac.authorization.k8s.io/v1
+    kind: Role
+    mode: propagate
+  - apiVersion: rbac.authorization.k8s.io/v1
+    kind: RoleBinding
+    mode: propagate
+  - apiVersion: v1
+    kind: Secret
+    mode: propagate
+  - apiVersion: v1
+    kind: ResourceQuota
+    mode: remove
+  - apiVersion: v1
+    kind: ConfigMap
+    mode: foobar`
```

Comment:
What should happen here? I think you could make the case that it should just be converted as-is, right?

Reply:
Ah I see, you mean still with `foobar` mode in the v1alpha2 and let the reconcilers decide on it (ignoring it).
Ah but we have the schema validation that it won't take anything other than `Ignore`, `Propagate` or `Remove`, so I guess we have to convert it and not leave it as-is. SGTY?

Reply:
Ah right, if the mode is wrong, converting it to Ignore probably makes sense.

What about if the apiVersion or kind are wrong?

Reply:
There's no schema validation on them yet (neither v1a1 or v1a2) but only webhooks to see if the type exists.

If the apiVersion or kind is wrong, the converted GR will not be found and will have `TypeNotFound` condition. Here's an current example of having `networking.k8s.io/v1/NetworkPolicy` converted to `networking.k8s.io/networkpolicys` (wrong resource, should be `networkpolicies`): 
```
spec:
  types:
  - group: networking.k8s.io
    mode: Propagate
    resource: networkpolicys
  - group: rbac.authorization.k8s.io
    mode: Propagate
    resource: roles
  - group: rbac.authorization.k8s.io
    mode: Propagate
    resource: rolebindings
status:
  conditions:
  - code: TypeNotFound
    msg: Resource "networkpolicys.networking.k8s.io" not found
  types:
  - group: rbac.authorization.k8s.io
    mode: Propagate
    numPropagatedObjects: 0
    numSourceObjects: 0
    resource: rolebindings
    version: v1
  - group: rbac.authorization.k8s.io
    mode: Propagate
    numPropagatedObjects: 0
    numSourceObjects: 0
    resource: roles
    version: v1
```

Reply:
sg, as long as we allow the "bad" type to be created with a condition rather than just omitting it. But we should definitely test that in a conversion test.

Reply:
sg done.